pkg/config/config.go

```diff
@@ -162,3 +151,67 @@ func ExportKubeProxyConfig(fullState *cluster.FullState, dataDir string) error {
 
 	return nil
 }
+
+func ConfigurePrivateRegistries(ctx context.Context, fullState *cluster.FullState, registriesTLS []string) error {
+	privateRegistryConfig := fullState.CurrentState.RancherKubernetesEngineConfig.PrivateRegistries
+	if len(privateRegistryConfig) <= 0 {
+		return nil
+	}
+	if _, err := os.Stat(privateRegistryPath); err != nil {
+		if os.IsNotExist(err) {
+			r := registries.Registry{}
+			r.Configs = make(map[string]registries.RegistryConfig)
+			for _, reg := range privateRegistryConfig {
+				endpoint := reg.URL
+				if endpoint == "" {
+					endpoint = "docker.io"
+				}
+				if reg.User != "" && reg.Password != "" {
```

Comment:
Could there ever be a circumstance where a user is set and no password?

Reply:
I dont believe thats possible with basic authentication for private registry